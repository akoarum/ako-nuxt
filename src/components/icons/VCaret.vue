<template>
  <span :style="styled" class="vCaret"></span>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    width: { type: Number },
    height: { type: Number }
  },
  computed: {
    styled(): { [index: string]: string } {
      const styles = {}

      if (this.width) {
        styles['--width'] = `${this.width}px`
      }

      if (this.height) {
        styles['--height'] = `${this.height}px`
      }

      return styles
    }
  }
})
</script>

<style lang="scss" scoped>
.vCaret {
  --width: 8px;
  --height: 6px;
  display: inline-block;
  border-width: var(--height) var(--width);
  border-style: solid;
  border-color: transparent transparent transparent currentColor;
}
</style>
