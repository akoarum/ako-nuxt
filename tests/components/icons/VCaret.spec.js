import { shallowMount } from '@vue/test-utils'
import VCaret from '~/components/icons/VCaret'

describe('VCaret', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VCaret, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
