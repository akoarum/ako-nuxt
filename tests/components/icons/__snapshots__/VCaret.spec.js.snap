// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCaret initialized snapshot 1`] = `
<span
  class="vCaret"
/>
`;
